mediacloud/mediawords/util/url/__init__.py

```diff
@@ -90,8 +90,18 @@ def is_http_url(url: str) -> bool:
     try:
         uri = furl(url)
 
-        # Some URLs have invalid paths that furl's constructor doesn't check
-        _ = str(uri.path)
+        # Try stringifying URL back from the furl() object to try out all of its accessors
+        _ = str(furl)
```

Comment:
should this still be `uri.path`?

Reply:
Thanks, fixed, it should have been `str(uri)`.

Stringifying the whole furl object covers more base, e.g. try:

```python
import furl

# Invalid URL which gets treated as IDNA for whatever reason
x = furl.furl('http://michigan-state-football-sexual-assault-charges-arrest-players-names')

# Prints out empty string -- bad, we want it to fail to learn that the URL is bogus
print(x.path)

# Raises -- good!
print(x)
```